File: prime_sieve/javascript/prime_sieve.js
const { range, compress } = require("./utils")

module.exports = {
    primeSieve: function(n){
       // Returns a list of primes < n for n > 2
       let sieve = new Array( Math.floor(n / 2) ).fill(1)
       for (var i = 3; i < parseInt(n**0.5, 10) + 1; i += 2 ){
          if (sieve[Math.floor(i / 2)]){
             for (var j = Math.floor(i*i/2); j < sieve.length; j += i){
                sieve[j] = 0;
             }      
          }
       }
       return [2].concat(compress(range(3, n, 2), sieve.slice(1,)))
    }
}

File: prime_sieve/python/prime_sieve.py

def prime_sieve(n):
    """ Returns a list of primes < n for n > 2 """
    sieve = bytearray([True]*(n//2))
    for i in range(3,int(n**0.5)+1,2):
        if sieve[i//2]:
            sieve[i*i//2::i] = bytearray((n-i*i-1)//(2*i)+1)
    return [2] + [ d for d,s in zip(range(3,n,2), sieve[1:]) if s ]

import time
start = time.time()
primes = prime_sieve(50000000)
print(f"{len(primes)} found in {time.time() - start} seconds.")
# 3001134 found in 1.7941629886627197 seconds. With input of 50,000,000